src/server/page_server.cpp

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2008 Search Solution Corporation
+ * Copyright 2016 CUBRID Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+#include "page_server.hpp"
+
+#include "error_manager.h"
+#include "log_impl.h"
+#include "log_prior_recv.hpp"
+#include "packer.hpp"
+#include "server_type.hpp"
+#include "system_parameter.h"
+
+#include <cassert>
+#include <functional>
+
+page_server ps_Gl;
+
+static void assert_page_server_type ();
+
+page_server::~page_server ()
+{
+  disconnect_active_tran_server ();
+}
+
+void
+page_server::set_active_tran_server_connection (cubcomm::channel &&chn)
+{
+  assert_page_server_type ();
+
+  chn.set_channel_name ("ATS_PS_comm");
+  er_log_debug (ARG_FILE_LINE, "Active transaction server connected to this page server. Channel id: %s.\n",
+		chn.get_channel_id ().c_str ());
+
+  m_ats_conn = new active_tran_server_conn (std::move (chn));
+
+  m_ats_conn->register_request_handler (ats_to_ps_request::SEND_LOG_PRIOR_LIST,
+					std::bind (&page_server::receive_log_prior_list, std::ref (*this),
+					    std::placeholders::_1));
+
+  m_ats_conn->start_thread ();
+}
+
+void
+page_server::disconnect_active_tran_server ()
+{
+  delete m_ats_conn;

```

Comment:
test if already connected? or if not already disconnected? or assert otherwise?

Reply:
May be both. If ats si connected, then it is deleted. If it is not connected, then m_ats_conn is nullptr and delete has no effect.

Reply:
I know it has no effect if already nullptr. Testing for nullptr before actual deletion clearly suggests intent. But it's just pedantic. :)